exercises/firmware_auditing/README.md

```diff
@@ -0,0 +1,318 @@
+<!--
+Copyright lowRISC Contributors.
+SPDX-License-Identifier: Apache-2.0
+-->
+# Firmware Auditing Exercise
+
+First, make sure to go to the [building the exercises][] section to see how the exercises are built.
+
+[building the exercises]: ../README.md#building-the-exercises
+
+Next, go to the [auditing firmware][] documentation section to get a brief introduction to `cheriot-audit` and the Rego policy language.
+Part 3 of the [hardware access control] exercise also shows a simple example of using CHERIoT Audit to ensure that compartments do not access MMIO capabilities they should not be accessing.
+
+[auditing firmware]: ../../doc/auditing-firmware.md
+[hardware access control]: ../hardware_access_control/README.md#part-3
+
+In this exercise, we use the `cheriot-audit` tool to audit the JSON firmware reports produced by the CHERIoT RTOS linker.
+This will let us assert a properties about our firmware images at link-time, guaranteeing desired safety checks.
+This exercise explores a set of self-contained policies which audit a variety of properties, to give an idea of what can be achieved using CHERIoT Audit.
+
+For this exercise, when the `xmake.lua` build file is mentioned, we are referring to [`exercises/firmware_auditing/xmake.lua`][].
+
+[`exercises/firmware_auditing/xmake.lua`]: ../../exercises/firmware_auditing/xmake.lua
+
+## A quick introduction to Rego
```

Comment:
This is a little heavy for the start of a tutorial. We want to get them playing. This section could be moved to the explanation in `doc/auditing-firmware`. Then you can nudge them to read this section with a link to it in the first sentence of part one.

Reply:
I agree. I've made this change, and added a little forewarning at the top of part 1 to tell them that Rego syntax might be unfamiliar and if they find themselves confused they should check the primer.